### ISSUE TITLE
ksonnet-lib installation

### ISSUE THREAD

**Issue by sh0rez** [grafana/tanka#100]
As experienced in #95, due to recent fixes to https://github.com/jsonnet-bundler/jsonnet-bundler it is not possible anymore to install single files of a package but only the whole package at once. 

However, we required this feature (which was actually a bug) to install https://github.com/ksonnet/ksonnet-lib (`k.libsonnet`). 

While `jb` is still able to install those libraries, they end up at another import path:

```diff
- import "k.libsonnet"
+ import "ksonnet.beta.3/k.libsonnet"
```

This is a breaking change for most users and most imported packages, as ksonnet previously magically allowed this import style which diverts from what `jsonnet-bundler` specified in its design doc.

I have no strong opinion how to tackle this issue yet, however here are some ideas around the topic for discussion:

1. **Explicit imports:** The primary problem seems to be that ksonnet introduced a kind-of global import (`k.libsonnet`), which equals to a orphaned file directly at the root of `vendor/`. However, `jb` takes a more explicit approach, where an import is usually named `<package>/<file>.libsonnet`. IMHO this is the better way, as it avoids such a confusion. Golang for example even took this idea further and included the packages origin as well (`<remote>/<package>/<file>.go`)
2. **Breaking change:** While it would be better if all users would switch a better import format, this won't happen in reality, at least not immediately. Changing this however would make migrating from `ks` to `tk` hard, as adopting Tanka would immediately break ksonnet. I feel like we must avoid thiss at any cost. 
3. **Tooling:** To do this, we could introduce some additional tooling:
3.1:  **Custom `jsonnet.Importer`**: a custom implementation could wrap `jsonnet.FileImporter` to alias `k.libsonnet` to the appropriate path at `vendor/`. In `-verbose` mode it could print warnings about the deprecated import style
3.2: **`tk lint`** (linter that doesn't exist yet): Could have a lint rule warning about the deprecated import
3.3: **`tk fix`** (code-rewriter that doesn't exist either): Could have a rule to automatically adapt all import paths

With such a combination (support of the deprecated format, plus tooling to help users switch to the new one) we could avoid a fully breaking change.

Everybody is invited and encouraged to discuss the above ideas to find the best possible solution :D 

/cc @rfratto @slim-bean @cyriltovena @tomwilkie @malcolmholmes @gouthamve @brancz @metalmatze @bgagnon @beorn7

**Comment by chancez** [comment]
Just hit this also. Any chance the docs can be updated at least?

**Comment by sh0rez** [comment]
Working on a big docs update, will be out around next week

**Comment by sh0rez** [comment]
The upcoming fix is described in #132 (now pinned). Closing this one, as we won't need more discussion on that topic
